### PR TITLE
DAOS-9894 control: Pass mem_size to engine when using emulated NVMe

### PR DESCRIPTION
**Pull request by tanabarr**
When scanning storage during daos_server start-up, AIO devices in bdev
storage tiers should be considered valid so that hugepage assignments
can be allocated and verified as normal, otherwise the engine process
will assume that DMA buffer memory requirements have not been met and
subsequently fail.

Signed-off-by: Tom Nabarro <tom.nabarro@intel.com>

### CODE REVIEW

**Comment by knard-intel** [comment]
Manually validated with DAOS docker platform.

File: src/control/server/storage/bdev.go [hunk 1]
```diff
@@ -331,6 +331,12 @@ func (ncs NvmeControllers) Update(ctrlrs ...*NvmeController) NvmeControllers {
 	return ncs
 }
 
+// NvmeAioDevice returns struct representing an emulated NVMe AIO device (file or kdev).
```
Comment (knard-intel) [inline src/control/server/storage/bdev.go:334]: Seems to be only used with storage/mocks.go => Should it not be better to put in the mock file ?

**Review by knard-intel — APPROVED** [review]
Seems fine for me and my understanding of its implication.

Manually validated with DAOS docker platform.

File: src/control/server/storage/bdev.go (hunk 1 above)
Comment (tanabarr) [reply src/control/server/storage/bdev.go:334]: these are intended to be used in backend in following PRs so I would prefer to keep them here, the idea is to improve the code quality and test coverage for using emulated NVMe and these types will help to facilitate that

Comment (knard-intel) [reply src/control/server/storage/bdev.go:334]: More than OK, if it could help to improve the code quality and coverage :)

**Comment by tanabarr** [comment]
This PR addresses the following situation: getCfgBdevs skipping any emulated NVMe devices and so when using emulation the check for hasBdevs returns false and the mem_size parameter that is normally passed to the engine was not being set. The fix was to not skip the emulated devices in the check and to make sure (which was already the case but I added a comment to clarify) that we don't fail if the emulated devices don't exist.

**Review by mjmac — APPROVED** [review]

**Review by mjmac — DISMISSED** [review]
Oops, when I reviewed this earlier I forgot to check the test coverage. Please re-run with:
`Features: control`

**Review by kjacque — APPROVED** [review]
Looks reasonable to me.

**Pull request merged**